### PR TITLE
Blob replacement and dropzone improvements

### PR DESCRIPTION
**Pull request by nmpcunha**

### CODE REVIEW

**Comment by guirenard** [comment]
:+1:

File: elements/nuxeo-document-actions/nuxeo-replace-blob-button.html [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+    <nuxeo-document id="doc" doc-id="[[document.uid]]" response="{{document}}" sync-indexing></nuxeo-document>
+
+    <dom-if if="[[_isAvailable(document)]]">
+      <template>
+        <div class="action" on-tap="_toggleDialog">
+          <paper-icon-button icon="[[icon]]" noink></paper-icon-button>
+          <span class="label" hidden$="[[!showLabel]]">[[_label]]</span>
+        </div>
+        <nuxeo-tooltip>[[_label]]</nuxeo-tooltip>
```
Comment (Gabez0r) [inline elements/nuxeo-document-actions/nuxeo-replace-blob-button.html]: There should be an id to the `paper-icon-button` use the `for` attribute on the tooltip pointing to that id. As it is right now, the tooltip will show up if the user hovers the dialog as well.

File: elements/nuxeo-document-actions/nuxeo-replace-blob-button.html [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+          path += (path === '' ? '' : '.' ) + pieces.shift();
+          if(Array.isArray(this.get(path, properties))) {
+            return path;
+          }
+        }
+        return null;
+      },
+
+      _replaceBlob: function() {
+        var rootProperty = this.xpath.includes('/')
```
Comment (Gabez0r) [inline elements/nuxeo-document-actions/nuxeo-replace-blob-button.html]: Please document why we need this root property. Right now it's clear, but later on some else may not understand its purpose.

File: elements/nuxeo-document-actions/nuxeo-replace-blob-button.html [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+          obj[path[0]] = {};
+        }
+        return this._createNestedObjectRecursive(obj[path[0]], path.slice(1));
+      },
+
+      _getRootProperty: function(pieces, properties) {
+        var path = '';
+        while(pieces.length > 0) {
+          path += (path === '' ? '' : '.' ) + pieces.shift();
+          if(Array.isArray(this.get(path, properties))) {
```
Comment (Gabez0r) [inline elements/nuxeo-document-actions/nuxeo-replace-blob-button.html]: Cleary this root property stops as soon as an array is found. Please document this method.

**Comment by Gabez0r** [comment]
👍

**Comment by nmpcunha** [comment]
👍

**Pull request merged**